olive/engine/footprint.py

```diff
@@ -0,0 +1,178 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import logging
+from collections import OrderedDict
+from datetime import datetime
+from pathlib import Path
+
+from olive.common.config_utils import ConfigBase, config_json_dumps, config_json_loads
+
+logger = logging.getLogger(__name__)
+
+
+class FootprintNodeMetric(ConfigBase):
+    value: dict
+    cmp_direction: dict = {}
+    is_goals_met: bool = False
+
+
+class FootprintNode(ConfigBase):
+    # None for no parent which means current model is the input model
+    parent_model_id: str = None
```

Comment:
>parent_model_id [](http://example.com/codeflow?start=4&length=15)

input_model_id is better or not?

Reply:
input_model_id may be same with the Olive engine input_model_id(torch_model)? I prefer to keep parent.